### ISSUE TITLE
"sizes" decreases, but does not increase, when viewport width changes

### ISSUE THREAD

**Issue by galbaras** [nosilver4u/ewww-image-optimizer#165]
**Describe the bug**
When making the viewport smaller, the "sizes" attribute of images goes down, but when making the viewport larger, it remains the same, and the images remain at the smallest width before growing the viewport.

**To Reproduce**
1. Go to https://www.noff.com.au/
2. Right-Click one of the content images and choose "Inspect"
3. Make the page pane narrower
4. Make the page pane widers

**Expected behavior**
The "sizes" attribute of images should match the area allocated for the display of the image.

**Screenshots**

**Additional context**
This was observed on Chrome with WebP active in the plugin.

**Comment by nosilver4u** [comment]
Hi @galbaras, that is already a known behavior and not correctable from our end. Once sizes is smaller AND a smaller image has been loaded, the img container has also become smaller. It is the size of the img container that is used to determine the sizes attribute.
Thus the fix is to make sure your CSS expands the img element to fill the available space, then lazysizes will have the information it needs to increase the value of the sizes attribute.

**Comment by galbaras** [comment]
Where is this being discussed?

**Comment by nosilver4u** [comment]
It's not something that's been publicly discussed to my knowledge. We discovered it while trying to track down another issue with the lazy loader (that will be fixed in the next release). In that case, the images were being scaled down more than necessary for retina-capable devices.

I suspect it's not very common for anyone to even notice this particular issue, because it generally requires one to change the window size back and forth to specifically to test the behavior. If it's a big enough issue for you, and you can't alter the CSS, the auto-scaling can be disabled with this filter:

`add_filter( 'eio_lazy_responsive', '__return_false' );`

It can also be disabled on specific images with the skip-autoscale class.

**Comment by galbaras** [comment]
Oh, I've handled this with CSS. I just wish I didn't have to, and not everybody is like me.

If this is a known issue, we should report it to the author of lazysizes. Where do we do that?

Also, going from portrait to landscape on a mobile device or tablet shouldn't be too uncommon. I'm sure people do this all the time.

**Comment by nosilver4u** [comment]
I already talked to the lazysizes author about it before I realized exactly what was going on. The trouble is that it can't be resolved from lazysizes, thus it isn't seen as an "issue". It's just a fundamental drawback of auto-calculating the sizes attribute based on the image size as rendered by the browser.

Theoretically, I think one could tell lazysizes to traverse up the DOM to find the appropriate parent/ancestor element on which to base the size. That does give me some ideas, so I'll leave this open until I've had a chance to look into them further.
